GabbyTam_Fatima_functionpractice.js

```diff
@@ -0,0 +1,238 @@
+
+/* This doesnt work because of the use of return, return stops/breaks the code
+
+
+function printLongestWord(arr){
+  for(let i=0; i<arr.length-1;i++){
+    if(arr[i].length> arr[i+1].length){
+      return arr[i];
+    } else {
+      return arr[i+1];
+    }
+  }
+}
+
+console.log(printLongestWord(["BoJack", "Princess", "Diane", "a", "Max", "Peanutbutter", "big", "blob"]));
+*/
+
+//this works but only for this set of strings, if the longest word comes before peanutbutter it is overwritten by peanutbutter in later computation 
+
+function printLongestWord(arr){
+  let longest= '';
+  for(let i=0; i<arr.length-1; i++){
+    if(arr[i].length> arr[i+1].length){
+
+      longest= arr[i];
+    } //else {
+    //continue;
+   // }
+  } 
+  return longest;
+}
+
+console.log(printLongestWord(["BoJack", "Princess", "Diane", "a", "Max", "Peanutbutter", "big", "blob"]));
+
+//Part 2: 3 Write a Javascript function called transmogrify. This function should accept three arguments, which you can assume will be numbers. Your function should return the "transmogrified" result.
+
+//The transmogrified result of three numbers is the product of the first two numbers, raised to the power of the third number.
+
+//For example, the transmogrified result of 5, 3, and 2 is (5 times 3) to the power of 2 is 225.
+
+const transmogrify = (num1, num2, num3) => {
+  let part1= num1*num2;
+  let part2= Math.pow(part1, num3);
+  return part2;
+}
+
+console.log(transmogrify(5, 3, 2));
+
+//Part 2: 4 Write a function that takes a parameter, a number. The function should print the Fibonacci sequence up to that number.
+
+//Adjust the function to push the even numbered values into an array.
+
+//Adjust the function to return the summed value of the array.
+
+
+
+const fibonacci= (number) => {
+  let start= 1;
+  let newNum= 0;
+  let fillArr= [];
+  let sum= 0;
+  for(let i=1; i<=number; i++){
+    //this code will only write the fibonacci numbers up until the number passed in as the argument 
+    //it has to be newNum PLUS start because newNum alone would go one computation too far, because the block of code runs after the if statement is checked 
+    if(newNum+start<=number){
+      newNum= newNum + start;
+      //need to update start or else it will stay at value of 1
+      //start has to be newnum-start because in the step above, start was added to newNum
+      start= newNum - start;
+      console.log(newNum);
+      if(newNum%2===0){
+      fillArr.push(newNum);
+      }
+    }
+  }
+  console.log(fillArr);
+  for (let j=0; j<=fillArr.length-1; j++){
+    //sum is initialized at 0, this loop and code will iterate through all elements in the fillArr and add the element to the sum
+    sum= sum+fillArr[j];
+  }
+  //console.log(sum) works, prints sum to the console 
+  if(sum<=4000000){
+  console.log(sum);
+  } else {
+    console.log('Sum exceeds 4 million');
+    console.log(sum);
+  }
+  //return does not work, not sure why :(
+  return sum;
+}
+
+fibonacci(5670);
```

Comment:
the return does work, it just doesn't get printed in the console! if you want to see the return value for a function you have to `console.log()` the function like so: `console.log(fibonacci(5670))`